ros_ws/src/depthai_ros/src/depth_map_publisher_node.cpp

```diff
@@ -15,42 +21,106 @@ DepthMapPublisherNode::DepthMapPublisherNode(
     oak_.reset(new DepthAI::DepthAI("", config_file_path_, false));
+ 
+    wls_filter_ = cv::ximgproc::createDisparityWLSFilterGeneric(false);
+    wls_filter_->setLambda(lambda);
+    wls_filter_->setSigmaColor(sigma);
 }
 
 // Destroying OAK-D ptr
 DepthMapPublisherNode::~DepthMapPublisherNode() { oak_->~DepthAI(); }
 
-// Depth map publisher
+// Depth map and landmark publisher
 void DepthMapPublisherNode::Publisher(uint8_t disparity_confidence_threshold)
 {
+    // setting the disparity confidence threshold
     oak_->send_disparity_confidence_threshold(disparity_confidence_threshold);
+    std::list<std::shared_ptr<NNetPacket>> op_NNet_detections;
+    std::cout << "publishing" << std::endl;
+
+    // creating lut to convert to depth map
+    std::vector<int> lut(256, 0);
+    float focal_length = 1280 / (2.f * std::tan(71.86 / 2 / 180.f * std::acos(-1)));
+    for (int i = 0; i < 256; ++i)
+    {
+        float z_m = (focal_length * 7.5 / i);
+        lut[i] = std::min(65535.f, z_m * 10.f);
+    }
 
     while (ros::ok())
     {
-        oak_->get_streams(output_streams_);  // Fetching the frames from the oak-d
+        cv::Mat rectified_right, filtered_disparity, disp_map;
+        // Fetching the frames and NN objects from the oak-d
+        oak_->get_streams(
+            output_streams_, op_NNet_detections);  
+
+        depthai_ros::detection detection_msg;
+
+        std::list<std::shared_ptr<NNetPacket>>::iterator it;
+        // iterating over list of NNetPacket and adding detections to ros message
+        for (it = op_NNet_detections.begin(); it != op_NNet_detections.end(); ++it)
+        {
+            auto obj_detections = (*it)->getDetectedObjects();
+            depthai_ros::detection temp_detection_msg;
+            for (int i = 0; i < obj_detections->detection_count; ++i)
+            {
+                auto detection = obj_detections->detections[i];
+                depthai_ros::BoundingBox object;
+                object.label_id =  detection.label ;
+                object.confidence = detection.confidence ;
+                object.xmin =  detection.x_min ;
+                object.ymin =  detection.y_min ;
+                object.xmax =  detection.x_max ;
+                object.ymax =  detection.y_max ;
+                object.depth.x = detection.depth_x ;
+                object.depth.y = detection.depth_y ;
+                object.depth.z = detection.depth_z ;
+                temp_detection_msg.detections.push_back(object);
+            }
+            detection_msg = temp_detection_msg;
+        }
+
+        // using wls_filter object to use rectified right 
+        // and disparity to create filtered disparity
+        cv::flip(*output_streams_["rectified_right"], rectified_right, 1);
+        output_streams_["disparity"]->convertTo(disp_map, CV_16S);
+        wls_filter_->filter(disp_map, rectified_right, filtered_disparity);
+
+        cv::Mat depth_map(720, 1280, CV_16UC1);
```

Comment:
Instead of hardcoding the dimensions, you should use `DepthAI::DefaultMonoHeight` and `DepthAI::DefaultMonoWidth` constants that you have in that class. Also, in other places in this PR